pkg/controller/constraint/constraint_controller.go

```diff
@@ -151,12 +191,38 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 			if err := r.Update(context.Background(), instance); err != nil {
 				return reconcile.Result{Requeue: true}, nil
 			}
+			// removing constraint entry from cache
+			delete(r.constraintsCache, constraintKindName)
 		}
 	}
+	r.updateConstraintsCache()
 
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileConstraint) updateConstraintsCache() {
```

Comment:
`updateConstraintsCache` is a misnomer as we are actually just reporting updated metrics from the constraints cache.

The name suggests the function changes constriantsCache, which it doesn't

Reply:
renamed to `reportTotalConstraints`

Reply:
LGTM